Assets/Scripts/DragDropToDistance.cs

```diff
@@ -0,0 +1,268 @@
+    private bool isClickedOn //returns true if raycast hits an item with the drag object script attached
+    {
+        get
+        {
+            Ray ray = playerCamera.ScreenPointToRay(currentScreenPosition);
+            RaycastHit hit;
+
+            if (Physics.Raycast(ray, out hit) )
+            {
+                return hit.transform == transform;
+            }
+
+            return false;
+        }
+    }
+
+    // Start is called before the first frame update
+    void Start()
+    {
+        //getting GameManager
+        gameManagerScript = GameObject.Find("GameManager").GetComponent<ShapeSorting_GameManager>();
+
+        //setting the target point in the Z axis for objects to be dragged into
+        playerCamera = Camera.main;
+        targetZ = playerCamera.transform.position.z + cameraDifferential;
+
+        //getting rigid body component
+        objectRB = GetComponent<Rigidbody>();
+
+    }
+
+    private void Awake()
+    {
+        //enabling interaction events
+        screenPosition.Enable();
+        press.Enable();
+
+        //to allow you to assess where on the screen has been clicked (I think)
+        screenPosition.performed += context => { currentScreenPosition = context.ReadValue<Vector2>(); };
+
+        //declaring what should happen press interaction starts
+        press.performed += _ =>
+        {
+
+            if (isClickedOn)
+            {
+                StartCoroutine(Drag());
+            }
+
+        };
+
+        //declaring what should happen press interaction ends
+        press.canceled += _ =>
+        {
+            isDragging = false;
+            yellowHighlight.SetActive(false);
+        };
+
+    }
+
+
+    //setting noTriggers as true as standard
+    //necessary to make the red 'wrong' highlight function properly
+    void FixedUpdate()
+    {
+        noTriggers = true;
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+        //turning off red highlight
+        if (noTriggers)
+        {
+            TurnOffRedHighlight();
+        }
+
+        if (isMoving && targetTransform != null)
+        {
+            // Calculate the direction towards the target
+            Vector3 direction = (targetTransform.position - transform.position).normalized;
+
+            // Calculate the distance to the target
+            float distance = Vector3.Distance(transform.position, targetTransform.position);
+
+            // Move towards the target with interpolation
+            transform.position += direction * moveSpeed * Time.deltaTime;
+
+            // Once the distance is very small, stop moving to middle
+            if (distance < 0.05f)
+            {
+                isAtMiddle = true;
+                isMoving = false;
+            }
+        }
+
+        //once middle is reached, move backwards
+        if (isAtMiddle)
+        {
+            MoveBackWard();
+        }
+
+    }
+
+    //move object backward and through hole
+    private void MoveBackWard()
```

Comment:
We can change the method name of this to 'MoveObjectThroughHole' which would remove the comment line above.